### PR TITLE
Merge DelayedStream's setError() into cancel()

### PR DESCRIPTION
**Pull request by zhangkun83**
DelayedClientTransport.PendingStream will override cancel(), which has a
clearer semantic.

### CODE REVIEW

File: core/src/main/java/io/grpc/internal/DelayedClientTransport.java [hunk 1]
```diff
@@ -257,7 +253,7 @@ void setError(Status reason) {
           }
         }
       }
-      super.setError(reason);
+      super.cancel(reason);
```
Comment (ejona86) [inline core/src/main/java/io/grpc/internal/DelayedClientTransport.java]: I should have noticed before, but we should cancel the stream before reporting transport terminated (so call super first).

File: core/src/main/java/io/grpc/internal/DelayedClientTransport.java [hunk 2]
```diff
@@ -163,7 +161,7 @@ public void shutdownNow(Status status) {
     }
     if (savedPendingStreams != null) {
       for (PendingStream stream : savedPendingStreams) {
-        stream.setError(status);
+        stream.cancel(status);
```
Comment (ejona86) [inline core/src/main/java/io/grpc/internal/DelayedClientTransport.java]: Status here would commonly be UNAVAILABLE, which is not permitted in ClientStream.cancel(). Let's just change ClientStream to allow anything other that `isOk()`?

**Comment by ejona86** [comment]
@zhangkun83 LGTM

File: core/src/main/java/io/grpc/internal/DelayedClientTransport.java (hunk 2 above)
Comment (zhangkun83) [inline core/src/main/java/io/grpc/internal/DelayedClientTransport.java]: Done.

File: core/src/main/java/io/grpc/internal/DelayedClientTransport.java (hunk 1 above)
Comment (zhangkun83) [inline core/src/main/java/io/grpc/internal/DelayedClientTransport.java]: Done.

**Pull request merged**